### ISSUE TITLE
xhr.js cannot support UTF-8 encoding content

### ISSUE THREAD

**Issue by shilin8805** [driftyco/cordova-plugin-wkwebview-engine#49]
In xhr.js use window.atob to decode base64 response content
but window.atob only support Latin1 range characters.
Is it possible to replace the window.atob to third base64 decoding tools, such as [webtoolkit.base64](http://www.webtoolkit.info/javascript-base64.html#.V9j4zJN97YU)?

**Comment by manucorporat** [comment]
You are probably right! Can you provide an example that fails with the current approach?

**Comment by shilin8805** [comment]
@manucorporat 

for example:
In bash shell，convert "中文" to base64 encoding，the result is：
```shell
$ echo 中文 | base64  
5Lit5paHCg==
```

In Chrome console，use window.atob to decode，the result is：
```javascript
> window.atob('5Lit5paHCg==')
ä¸­æ
```

Use third base64 decoding tools in Chrome console，such as webtoolkit.base64，the result is:
```javascript
> Base64.decode('5Lit5paHCg==');
中文
```

**Comment by manucorporat** [comment]
@shilin8805 thanks a lot! I am going to work on this issue today

**Comment by aemr3** [comment]
@manucorporat any news on this? We are having same issue.

**Comment by manucorporat** [comment]
@aemr3 @shilin8805 fixed 6580819 !!!
thanks a lot for testing and reporting WK issues!

Before:
![screen shot 2016-09-18 at 13 59 54](https://cloud.githubusercontent.com/assets/127379/18615472/5f3a6df4-7da8-11e6-97eb-10a5e8cb59fa.png)

After:
![screen shot 2016-09-18 at 13 52 01](https://cloud.githubusercontent.com/assets/127379/18615471/5f3777c0-7da8-11e6-90ad-4507a60656e0.png)

**Comment by aemr3** [comment]
Thank you @manucorporat. Your last commit fixes the problem. We can now take the advantage of WK.
